### ISSUE TITLE
 Merging Synthetic Identities

### ISSUE THREAD

**Issue by ken-non** [department-of-veterans-affairs/va.gov-team#36326]
Spike to determine what actions to be taken (if any)

**Task is done when:**
 - Research is present in confluence.

### Notes
 - "I have an account in github and one in login.gov"
 - How do we see the same thing no matter how l log in?
 - How do I see the same thing logging in with login.gov as i would with github
 - How to handle folks without piv / etc?
 - another convo with Thomas
 - Set up a github app config - locally
 - integrate it with keycloak  - locally
 - Need to associate two IdPs with one principal
 - Document Output of spike
 - Just as easy as adding another IdP to keycloak?
 - Does it matter if they use different emails? (github vs logn.gov)
 - How to handle groups in github?
 - Make sure we can do this?
 - does it happen "out of the box"? Might be that easy

**Comment by kylesoskin** [comment]
Here are some of the findings: 
https://vfs.atlassian.net/wiki/spaces/AMT/pages/2089222145/Account+Federation+in+Keycloak+Notes

Tl;dr
If emails (by default) match between multiple idPs they can be automatically connected in keycloak. However by default the user is prompted on login if they want to link accounts. This can be changed to be automatic with a drop-down.

**Comment by kylesoskin** [comment]
@kjduensing Had requested I dive deeper on account flows in the doc, but I ended up linking the actual keycloak documentation since I was just copy/pasting snippets out of there anyway. He said this was sufficient and also asked me to test out the auto linking which I will outline here and then close this ticket out at standups. 
Account with email set.
![Screen Shot 2022-02-17 at 7.55.26 AM.png](https://images.zenhubusercontent.com/62067253500f33af56810776/500deb58-8204-4c5c-ab7e-8e472ca2a0d6)
It has no idPs
![image.png](https://images.zenhubusercontent.com/62067253500f33af56810776/440c4bd1-d88c-4345-a632-46324794c6b3)
Go to the idP section
![image.png](https://images.zenhubusercontent.com/62067253500f33af56810776/8f3914be-9b90-44ab-82df-18f3743d8e03)
![Screen Shot 2022-02-17 at 8.00.40 AM.png](https://images.zenhubusercontent.com/62067253500f33af56810776/2fed1d24-ebcb-427f-a727-9db4e3334b5d)
Disable "confirmation page" 
![Screen Shot 2022-02-17 at 8.11.25 AM.png](https://images.zenhubusercontent.com/62067253500f33af56810776/6d0ed07d-11ce-4808-9a6e-bf88537d05c3)
